File: 03_hash/sha256/hex/main.go
package main

import (
	"crypto/sha256"
	"encoding/hex"
	"fmt"
)

func main() {
	text := "Hello world, mac was here"
	hasher := sha256.New()
	hasher.Write([]byte(text))
	hash := hex.EncodeToString(hasher.Sum(nil))
	fmt.Println(hash)
}

File: 03_hash/sha256/hex/main.java
import java.math.BigInteger;
import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

class Main {
    public static void main(String[] args) throws NoSuchAlgorithmException {
        String text = "Hello world, mac was here";
        String hash = new BigInteger(1, MessageDigest.getInstance("SHA-256").digest(text.getBytes(StandardCharsets.UTF_8))).toString(16);
        System.out.println(hash);
    }
}
